pkg/telemetery/settings.go

```diff
@@ -14,11 +16,12 @@ import (
 )
 
 type Setting struct {
-	Logger         *zap.Logger
-	TracerProvider trace.TracerProvider
-	Metrics        metrics.Factory
-	ReportStatus   func(*componentstatus.Event)
-	Host           component.Host
+	Logger               *zap.Logger
+	TracerProvider       trace.TracerProvider
+	Metrics              metrics.Factory
+	ReportStatus         func(*componentstatus.Event)
+	Host                 component.Host
+	LeveledMeterProvider func(_ configtelemetry.Level) metric.MeterProvider
```

Comment:
Now that we've got this field available to us - we also want to be able to use it. For example, in the [query server ](https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/server.go#L153)we currently hardcode the meter provider but now we can pass it from `telset`.